src/dst/src/Distributed.cc

```diff
@@ -91,11 +91,13 @@ void Distributed::runWorker(const char* ip,
   }
 }
 
-void Distributed::runLoadBalancer(const char* ip, unsigned short port)
+void Distributed::runLoadBalancer(const char* ip, unsigned short port, const char* workersDNS)
 {
   try {
     asio::io_service io_service;
     LoadBalancer balancer(this, io_service, logger_, ip, port);
+    if(!lookUpWorkers(workersDNS, port))
```

Comment:
Do you already identify the first message? Is it the message of type "JobMessage::BALANCER"? Maybe instead of starting workers discovery later, we can start the discovery thread that you mention later with the start of the balancer?

Reply:
There is no first message identification for now. But I agree with the discovery thread.